src/nni_manager/training_service/dlts/dltsTrainingService.ts

```diff
@@ -0,0 +1,473 @@
+    private readonly jobQueue: string[];
+    private stopping: boolean = false;
+    private readonly experimentId!: string;
+    private versionCheck: boolean = true;
+    private logCollection: string = 'none';
+    private isMultiPhase: boolean = false;
+    private dltsRestServerPort?: number;
+
+    private readonly trialJobsMap: Map<string, DLTSTrialJobDetail>;
+    private nniManagerIpConfig?: NNIManagerIpConfig;
+    private dltsClusterConfig?: DLTSClusterConfig;
+    private dltsTrialConfig?: DLTSTrialConfig;
+
+    constructor() {
+        this.log = getLogger();
+        this.metricsEmitter = new EventEmitter();
+        this.trialJobsMap = new Map();
+        this.jobQueue = [];
+        this.experimentId = getExperimentId();
+        this.log.info('Construct DLTS training service.');
+    }
+
+    public async run(): Promise<void> {
+        this.log.info('Run DLTS training service.');
+        const restServer: DLTSJobRestServer = component.get(DLTSJobRestServer);
+        await restServer.start();
+        restServer.setEnableVersionCheck = this.versionCheck;
+        this.log.info(`DLTS Training service rest server listening on: ${restServer.endPoint}`);
+        await Promise.all([
+            this.statusCheckingLoop(),
+            this.submitJobLoop()]);
+        this.log.info('DLTS training service exit.');
+    }
+
+    private async statusCheckingLoop(): Promise<void> {
+        while (!this.stopping) {
+            const updateDLTSTrialJobs: Promise<void>[] = [];
+            for (const [trialJobId, dltsTrialJob] of this.trialJobsMap) {
+                updateDLTSTrialJobs.push(this.getDLTSTrialJobInfo(dltsTrialJob));
+            }
+    
+            await Promise.all(updateDLTSTrialJobs);
+
+            const restServer: DLTSJobRestServer = component.get(DLTSJobRestServer);
+            if (restServer.getErrorMessage !== undefined) {
+                throw new Error(restServer.getErrorMessage);
+            }
+            await delay(3000);
+        }
+    }
+
+    private async getDLTSTrialJobInfo(dltsTrialJob: DLTSTrialJobDetail): Promise<void> {
+        if (this.dltsClusterConfig == null) {
+            throw Error('Cluster config is not set');
+        }
+        const requestOptions: request.Options = {
+            uri: `${this.dltsClusterConfig.dashboard}api/v2/clusters/${this.dltsClusterConfig.cluster}/jobs/${dltsTrialJob.dltsJobId}`,
+            qs: {
+                email: this.dltsClusterConfig.email,
+                password: this.dltsClusterConfig.password
+            },
+            json: true
+        };
+        const body = await new Promise((resolve, reject) => {
+            request(requestOptions, (error, response, body) => {
+                if (error != null) {
+                    reject(error)
+                } else {
+                    resolve(body)
+                }
+            })
+        }) as any;
+        switch (body['jobStatus']) {
+            case 'unapproved':
```

Comment:
what is the difference between 'unapproved' and 'failed', I mean from the consequence. for 'queued' and 'scheduling', at least the job will run sometime.

Reply:
'unapproved' is the state at the very beginning?

Reply:
`unapproved`: user applied for resources more than their quota, will go to `queued` after admin approve the job.
`queued`: waiting for available resource.
`scheduling`: k8s pod created, pulling images or running init scripts.

`failed` just means the job container exited with non-0

Reply:
got it, then 'waiting' is fine